docs/functional-areas/crtp/index.md

```diff
@@ -3,69 +3,137 @@ title: CRTP - Communication with the Crazyflie
+The version is available using the [getProtocolVersion](crtp_platform#get-protocol-version) packet.
 
- | Carrier          | Supports|
- | -----------------| -------------------|
- | Crazyradio (PA)  | Crazyflie 1.0/2.X|
- | USB              | Crazyflie 2.X|
+The version is currently 5.
 
-Header
-------
+When removing functionality from the protocol, packet will be depreciated for at least one version before being removed.
+Depreciated functionality prints a Warning in the [console](crtp_console). This rule allows for the Crazyflie firmware
+to evolve and remove old functionalities when needed and for a client or lib to have some compatibility guarantee.
+For example if a client support the protocol version 5, it is guarantee to work with version 6 but should likely
+not accept to connect a Crazyflie on version 7 of the protocol.
 
-Each packet has a 1 byte header and can carry up to 31 bytes data
-payload. The header field has the following layout:
+## Communication stack
 
-       7    6    5    4    3    2    1    0
-    +----+----+----+----+----+----+----+----+
-    |       Port        |  Link   |  Chan.  |
-    +----+----+----+----+----+----+----+----+
+The Crayzyflie communication is implemented as a stack of independent layers:
 
-Where:
+    +-------------------+
+    +     Subsystems    +   <- Log/Param/Commander/...
+    +-------------------+
+    +       CRTP        +   <- (port, channel, payload)
+    +-------------------+
+    +       Link        +   <- Radio link or USB link
+    +-------------------+   
+    +  Physical medium  +   <- radio or USB
+    +-------------------+
 
--   **Port**: Used to identify the functionality or task that is
-    associated with this message
--   **Link**: Reserved for future use
--   **Channel**: Used to identify the sub-task/functionality
+ - The **physical layer** is responsible for transmitting packet to and from the
+   Crazyflie. Currently USB and Radio are actively supported.
+ - The **link**  is responsible of implementing safe and ordered packet channels
+   to and from the Crazyflie. The link abstract the physical medium and
+   implement one transmit and one receive packet channels to and from the
+   Crazyflie.
+ - **CRTP** implements *port* and *channel* information used to route the packet
+   to various subsystem. *port* are very similar to UDP port, they are assigned
+   to subsystems. *channel* are intended to be used in subsystems to route
+   packets to different functionalities.
+ - **Subsystems** implements the Crazyflie functionalities that can be
+   controlled over CRTP. There is one *port* assigned to each *subsystem*.
 
-Port allocation
----------------
 
-Current port allocation:
+## Link implementations
+
+There is currently two actively supported link implementation. They are
+documented on their own thread:
+ - [The radio link]() implements CRTP link over nRF24 compatible radios
+ - [The USB link]() implements CRTP link over USB to the Crazylfie 2.x USB port 
```

Comment:
These links now to the same page it is now. A bit confusing as I would expect to either go to a subtitle or a different page.

Reply:
Those are not documented yet. I remove the link.